test/core/end2end/fixtures/http_proxy_fixture.cc

```diff
@@ -610,12 +612,9 @@ grpc_end2end_http_proxy* grpc_end2end_http_proxy_create(
   proxy->proxy_name = grpc_core::JoinHostPort("localhost", proxy_port);
   gpr_log(GPR_INFO, "Proxy address: %s", proxy->proxy_name.c_str());
   // Create TCP server.
-  proxy->channel_args = grpc_channel_args_copy(args);
-  grpc_error_handle error = grpc_tcp_server_create(
-      nullptr, proxy->channel_args,
-      grpc_slice_allocator_factory_create(
-          grpc_resource_quota_from_channel_args(args, true)),
-      &proxy->server);
+  proxy->channel_args = grpc_core::EnsureResourceQuotaInChannelArgs(args);
```

Comment:
If this function does not have ownership of the args, then my suggestion about ownership transfer in EnsureResourceQuotaInChannelArgs doesn't hold. I think this is an uncommon case here, though.